### PR TITLE
[UOE] Update order sync failed event

### PR DESCRIPTION
**Pull request by atorresveiga**
Part of: #6711

### Description
This PR adds the `flow` parameter to order sync failed event:

- order_sync_failed

### Testing instructions
TC1

1. Go to Orders
2. Select an Order -> press Edit
3. Turn `Plain mode` ON in the device
4. Made a change in the order
5. Check that `🔵 Tracked: order_sync_failed, Properties: {"flow":"editing",...}` is triggered

TC2

1. Go to Orders
2. Tap (+) Add Button -> Create order
3. Turn `Plain mode` ON in the device
4. Add a product to the order
5. Check that `🔵 Tracked: order_sync_failed, Properties: {"flow":"editing",...}` is triggered

- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/creation/UnifiedOrderEditViewModelTest.kt [hunk 1]
```diff
@@ -172,6 +177,28 @@ abstract class UnifiedOrderEditViewModelTest : BaseUnitTest() {
...
+        )
+        val throwable = WooException(error = wooError)
+        createUpdateOrderUseCase = mock {
+            onBlocking { invoke(any(), any()) } doReturn flowOf(Failed(throwable))
+        }
+        createSut()
+
+        verify(tracker).track(
+            stat = AnalyticsEvent.ORDER_SYNC_FAILED,
+            properties = mapOf(AnalyticsTracker.KEY_FLOW to tracksFlow),
```
Comment (atorresveiga) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/creation/UnifiedOrderEditViewModelTest.kt:294]: @wzieba, I could use some help with this test. Somehow the `flow` field in OrderCreateEditViewModel is null when the test reaches the tracking method causing this test to fail 😢

Comment (wzieba) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/creation/UnifiedOrderEditViewModelTest.kt:294]: 👋 I see the issue was related to the order of declaring `flow` String and declaring methods. I think this was happening before:

> If any of the properties are accessed before they are initialized w.r.t initialization order (e.g., if a method called in an initialization block accesses a property declared after the initialization block), the value of the property is unspecified. It stays unspecified even after the “proper” initialization is performed.

(from this [SO answer](https://stackoverflow.com/a/71487047), citing Kotlin docs).

I fixed this in ca8a6d1695421bf68db99cc62f71ce57b13405f9. I always thought that all properties are initialized at the same time as constructor (just after the constructor), TIL they're not and the order matters.

---

The tests are still not passing for `edit` case, I think there's an issue with mocking. Could you take a look? 🙏

Comment (atorresveiga) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/creation/UnifiedOrderEditViewModelTest.kt:294]: Thanks @wzieba !!🎉

>Could you take a look? 

On it 👀

**Review by nbradbury — APPROVED** [review]
Looks good! :shipit:

**Pull request merged**